### PR TITLE
Enhancement - generate confirm table data from steps

### PR DESCRIPTION
**Pull request by joefitter**
- added logic to generate an array of sections containing fields for display on confirm page and email
- updated readme
- added tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -300,6 +286,16 @@ In config page template
 {{/tableSections}}
 ```
 
+This assumes all steps containing fields have a `section` - `locals.section`, this is used to group fields in the confirm table and the email.
```
Comment (JoeChapman) [inline README.md:289]: Does this mean all steps containing fields _must_ have a section?

Comment (joefitter) [reply README.md:289]: all of our services do, they need it for displaying a title.  I guess we could improve it by catering for step config without sections too, but think that should probably be a further iteration

Comment (JoeChapman) [reply README.md:289]: I just thought that perhaps you could make more of a point about it and a little code example.

Comment (joefitter) [reply README.md:289]: It's a good point though - it should be able to handle lack of sections, we introduced that so probably not relevant for all apps. I'll add a code example for now, but we should change this so it supports apps without sections

**Pull request merged**